nav2-usability/scenario-code/login-routing/router.dart

```diff
@@ -0,0 +1,253 @@
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(BooksApp());
+}
+
+class BooksApp extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() => _BooksAppState();
+}
+
+class AppState extends ChangeNotifier {
+  bool _isLoggedIn = false;
+
+  bool get isLoggedIn => _isLoggedIn;
```

Comment:
The login flow should not be in the app state. It should a wrapper over a (fake) http client.

Depends on how complex/realistic we want this to be, we may store some session token in the appstate.

Reply:
fixed - I added an Authentication class implemented by a MockAuthentication class